### ISSUE TITLE
(aws-stepfunctions-tasks): StepFunctionsStartExecution createScopedAccessPolicy() assumes current account

### ISSUE THREAD

**Issue by pawelzebrowski** [aws/aws-cdk#19206]
### Description

StepFunctionsStartExecution creates policy statements as part of `createScopedAccessPolicy()` when Integration Pattern is RUN_JOB.  This does not support creating SFN definitions that will be deployed cross-account using AWS DeployPipeline because the policy statements assume current account.

### Use Case

Define and write an SFN using CDK that will be deployed cross account using CodePipeline

### Proposed Solution

Allow optional `account` parameter in `StepFunctionsStartExecutionProps` which is used during policy creation.  `stack.formatArn()` accepts an account property that can be used.

### Acknowledge

- [X] I may be able to implement this feature request
- [ ] This feature might incur a breaking change

**Comment by kaizen3031593** [comment]
Are you talking about cross account state machines using `fromStateMachineArn`? If so, I should have fixed this issue in this [PR](https://github.com/aws/aws-cdk/pull/19026) which released in v1.145.0. `fromStateMachineArn` now does not assume the current account and rather gets the environment from the arn.

If you're talking about something else though, can you provide a code example?

**Comment by pawelzebrowski** [comment]
I don't think that PR addresses the cross account issue i'm facing.

In the interim, i solved this by forking a local copy of StepFunctionsStartExecution with two changes.  This allows me to pass in an account if needed.  I haven't studied the side effects in detail, but if benign then it would be helpful to allow this optional parameter natively.

```
export interface StepFunctionsStartExecutionProps extends sfn.TaskStateBaseProps {
  ...

  /**
   * The account to use for any scoped access policy creation.  Used in multi-account codepipeline scenarios
   *
   * @default - None
   */
  readonly account?: string;
}

...

  private createScopedAccessPolicy(): iam.PolicyStatement[] {
    const stack = Stack.of(this);

    const policyStatements = [
      new iam.PolicyStatement({
        actions: ['states:StartExecution'],
        resources: [this.props.stateMachine.stateMachineArn],
      }),
    ];

    // Step Functions use Cloud Watch managed rules to deal with synchronous tasks.
    if (this.integrationPattern === sfn.IntegrationPattern.RUN_JOB) {
      policyStatements.push(
        new iam.PolicyStatement({
          actions: ['states:DescribeExecution', 'states:StopExecution'],
          // https://docs.aws.amazon.com/step-functions/latest/dg/concept-create-iam-advanced.html#concept-create-iam-advanced-execution
          resources: [
            stack.formatArn({
              service: 'states',
              account: this.props.account, // <---- specify account if passed in
              resource: 'execution',
              arnFormat: ArnFormat.COLON_RESOURCE_NAME,
              resourceName: `${
                stack.splitArn(
                  this.props.stateMachine.stateMachineArn,
                  ArnFormat.COLON_RESOURCE_NAME
                ).resourceName
              }*`,
            }),
          ],
        })
      );

      policyStatements.push(
        new iam.PolicyStatement({
          actions: [
            'events:PutTargets',
            'events:PutRule',
            'events:DescribeRule',
          ],
          resources: [
            stack.formatArn({
              service: 'events',
              account: this.props.account, // <---- specify account if passed in
              resource: 'rule',
              resourceName:
                'StepFunctionsGetEventsForStepFunctionsExecutionRule',
            }),
          ],
        })
      );
    }

    return policyStatements;
  }

```
